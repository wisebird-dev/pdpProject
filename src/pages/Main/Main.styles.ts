import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles({
    phonesContainer: {
        display: 'flex',
        flexDirection: 'row',
        flexWrap: 'wrap',
        alignItems: 'flex-start',
        justifyContent: 'center',
    },
});

export default useStyles;
