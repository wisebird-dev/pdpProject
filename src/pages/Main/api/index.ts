export { GET_SERVER_INFO } from './gql';
