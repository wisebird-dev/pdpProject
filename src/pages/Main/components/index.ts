export { Head } from './Head';
export { StructuredDataScript } from './Head/components/StructuredDataScript';
