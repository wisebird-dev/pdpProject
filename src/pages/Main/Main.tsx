import React, { useEffect } from 'react';
import { useQuery } from '@apollo/client';
import { Head } from './components';

import { useAppDispatch } from '@hooks/store';
import { setRunningAction } from '@store/actions';
import { Typography } from '@components';

import { GET_SERVER_INFO } from './api';

export interface MainProps {
    name?: string;
}

const Main: React.FC<MainProps> = ({ name }: MainProps) => {
    const dispatch = useAppDispatch();
    const { loading, error, data } = useQuery(GET_SERVER_INFO);

    useEffect(() => {
        dispatch(setRunningAction({ running: true }));
    }, []);

    return (
        <>
            <Head />
            <h1>PDP project</h1>
            <h2>{name ?? 'name'}</h2>
            <h2 style={{ color: 'orange' }}>Hello World!!</h2>
            <h2>{loading ? 'loading' : 'finished loading'}</h2>
            <h2>{error ? `error: ${error.message}` : 'no errors'}</h2>
            <Typography variant="body1" color="textPrimary">
                server message: {data ? data.info : 'loading'}
            </Typography>
        </>
    );
};

export default Main;
