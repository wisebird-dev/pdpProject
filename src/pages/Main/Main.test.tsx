import React from 'react';
import { renderWithProviders } from '@testUtils';

import Main from './Main';

describe('Main page suite', () => {
    it('renders hello world message', () => {
        const { getByText } = renderWithProviders(<Main />);
        const expectedText = 'Hello World!!';

        expect(getByText(expectedText)).toBeInTheDocument();
    });
});
