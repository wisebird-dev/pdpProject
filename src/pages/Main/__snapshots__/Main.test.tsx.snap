// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page suite matches snapshot 1`] = `
<div
  class="MuiContainer-root makeStyles-container-3"
>
  <div
    class="MuiBox-root MuiBox-root-7"
  >
    <div
      class="MuiBox-root MuiBox-root-8"
    >
      <h3
        class="MuiTypography-root makeStyles-productName-4 MuiTypography-h2"
      >
        iPhone 13 Pro
      </h3>
    </div>
    <div
      class="MuiBox-root MuiBox-root-9"
    >
      <h2
        class="MuiTypography-root makeStyles-productMoto-5 MuiTypography-h4"
      >
        Oh.So.Pro
      </h2>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-10 makeStyles-imageContainer-6"
  >
    <div
      style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
    >
      <noscript />
      <img
        alt="iphone 13 Pro"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
    </div>
  </div>
</div>
`;
